### PR TITLE
feat: Generic Event Source Mapping Trigger support

### PR DESCRIPTION
**Pull request by jvasallo**
Solves #355 and #201 

In addition adds support for SQS

### CODE REVIEW

**Comment by sijis** [comment]
Still WIP?

**Comment by sijis** [comment]
@jvasallo Are there any gotcha's with this or anything else left to address?

**Comment by jvasallo** [comment]
Waiting on @e4r7hbug to confirm the review is good. :)

File: src/foremast/awslambda/dynamodb_streams_event/dynamodb_streams_event.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        env (str): Environment/Account for lambda function
+        region (str): AWS region of the lambda function
+        rules (str): Trigger rules from the settings
+    """
+    session = boto3.Session(profile_name=env, region_name=region)
+    lambda_client = session.client('lambda')
+
+    trigger_arn = rules.get('stream')
+    if not trigger_arn:
+        trigger_arn = rules.get('table')
```
[the remark this answers is not in the document]
Comment (jvasallo) [reply src/foremast/awslambda/dynamodb_streams_event/dynamodb_streams_event.py]: Added a check and raise if we still do not get a trigger_arn

File: src/foremast/awslambda/dynamodb_stream_event/dynamodb_stream_event.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        env (str): Environment/Account for lambda function
+        region (str): AWS region of the lambda function
+        rules (str): Trigger rules from the settings
+    """
+    session = boto3.Session(profile_name=env, region_name=region)
+    lambda_client = session.client('lambda')
+
+    trigger_arn = rules.get('stream')
+    if not trigger_arn:
+        trigger_arn = rules.get('table')
```
[the remark this answers is not in the document]
Comment (jvasallo) [reply src/foremast/awslambda/dynamodb_stream_event/dynamodb_stream_event.py]: Raising an exception.

File: src/foremast/utils/dynamodb_stream.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    dynamodb_streams_client = session.client('dynamodbstreams')
+
+    if arn_string:
+        table_name = arn_string.split(':')[-1].split('table/')[-1]
+    LOG.info(table_name)
+    table_response = dynamodb_client.describe_table(TableName=table_name)
+    table_info = table_response['Table']
+
+    if not table_info:
+        LOG.critical("No DynamoDB table named %s found.", table_name)
```
[the remark this answers is not in the document]
Comment (jvasallo) [reply src/foremast/utils/dynamodb_stream.py]: You are right. Fixing.

File: src/foremast/utils/dynamodb_stream.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    """
+    session = boto3.Session(profile_name=account, region_name=region)
+    dynamodb_client = session.client('dynamodb')
+    dynamodb_streams_client = session.client('dynamodbstreams')
+
+    if arn_string:
+        table_name = arn_string.split(':')[-1].split('table/')[-1]
+    LOG.info(table_name)
+    table_response = dynamodb_client.describe_table(TableName=table_name)
```
[the remark this answers is not in the document]
Comment (jvasallo) [reply src/foremast/utils/dynamodb_stream.py]: Agreed. Adding some better logging here.

File: src/foremast/awslambda/dynamodb_stream_event/dynamodb_stream_event.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    lambda_alias_arn = get_lambda_alias_arn(app=app_name, account=env, region=region)
+    stream_arn = get_dynamodb_stream_arn(arn_string=trigger_arn, account=env, region=region)
```
[the remark this answers is not in the document]
Comment (jvasallo) [reply src/foremast/awslambda/dynamodb_stream_event/dynamodb_stream_event.py]: Good point. Bail out earlier than later

File: src/foremast/utils/dynamodb_stream.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    if arn_string.startswith('arn:aws:dynamodb:'):
+        _prefix, table, *stream = arn_string.split('/')
+
+        LOG.debug("ARN Split: %s %s %s", _prefix, table, *stream)
+        if stream:
+            arn_type = "dynamodb-stream"
+        else:
+            arn_type = "dynamodb-table"
+
+    return arn_type
```
[the remark this answers is not in the document]
Comment (jvasallo) [reply src/foremast/utils/dynamodb_stream.py]: Good catch!

**Comment by jvasallo** [comment]
Going to do a quick wave of checking this feature. Adding WIP until that is done.

**Pull request merged**